app/src/main/kotlin/com/wire/android/ui/userprofile/UserProfileViewModel.kt

```diff
@@ -14,48 +14,49 @@ import com.wire.android.navigation.NavigationCommand
 import com.wire.android.navigation.NavigationItem
 import com.wire.android.navigation.NavigationManager
 import com.wire.android.util.extension.toByteArray
+import com.wire.kalium.logic.feature.user.GetSelfUserUseCase
 import com.wire.kalium.logic.feature.user.UploadUserAvatarUseCase
+import com.wire.kalium.logic.functional.Either
 import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
 import kotlinx.coroutines.Dispatchers
-import com.wire.kalium.logic.functional.Either
 import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
-import javax.inject.Inject
 
-
-//Suppress for now after removing mockMethodForAvatar it should not complain
-@Suppress("TooManyFunctions","MagicNumber")
+// Suppress for now after removing mockMethodForAvatar it should not complain
+@Suppress("TooManyFunctions", "MagicNumber")
 @ExperimentalMaterial3Api
 @HiltViewModel
 class UserProfileViewModel @Inject constructor(
     private val navigationManager: NavigationManager,
     private val dataStore: UserDataStore,
-    private val uploadUserAvatarUseCase: UploadUserAvatarUseCase
+    private val uploadUserAvatar: UploadUserAvatarUseCase,
+    private val getSelf: GetSelfUserUseCase
 ) : ViewModel() {
 
-    //This is going to be refactored and removed later on
+    // This is going to be refactored and removed later on
     private val tempBitmapWidth = 36
     private val tempBitmapHeight = 36
 
-    var userProfileState by mutableStateOf(
-        SelfUserProfileState(
-            avatarBitmap = Bitmap.createBitmap(tempBitmapWidth, tempBitmapHeight, Bitmap.Config.ARGB_8888),
-            status = UserStatus.BUSY,
-            fullName = "Tester Tost_long_long_long long  long  long  long  long  long ",
-            userName = "@userName_long_long_long_long_long_long_long_long_long_long",
-            teamName = "Best team ever long  long  long  long  long  long  long  long  long ",
-            otherAccounts = listOf(
-                OtherAccount("someId", "", "Other Name 0", "team A"),
-                //                OtherAccount("someId", "", "Other Name 1", "team B"),
-                //                OtherAccount("someId", "", "Other Name 2", "team C"),
-                //                OtherAccount("someId", "", "Other Name", "team A"),
-                OtherAccount("someId", "", "New Name")
-            )
-        )
-    )
+    var userProfileState by mutableStateOf(SelfUserProfileState())
         private set
 
+    init {
+        viewModelScope.launch {
+            getSelf().collect {
+                userProfileState = SelfUserProfileState(
+                    avatarBitmap = null,
+                    status = UserStatus.AVAILABLE,
+                    fullName = it.name!!,
+                    userName = it.handle!!,
+                    teamName = it.team ?: "",
+                    otherAccounts = emptyList()
+                )
+            }
+        }
+    }
```

Comment:
we are not handeling here anywhere loading of the data as well as handeling error, do not know if that is something we should do for now, if not maybe add TODO: to stuff that we need to adjust later on

Reply:
Thanks! yes, definitely we should have some initial state for loading and play with that. Will add a TODO for in the meanwhile